CHANGELOG.md

```diff
@@ -23,6 +23,11 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Both the `Histogram` and `HistogramDataPoint` are redefined with a generic argument of `[N int64 | float64]` in `go.opentelemetry.io/otel/sdk/metric/metricdata`. (#3849)
 - The metric `Export` interface from `go.opentelemetry.io/otel/sdk/metric` accepts a `*ResourceMetrics` instead of `ResourceMetrics`. (#3853)
 
+### Fixed
+
+- Fix race conditions in `go.opentelemetry.io/otel/exporters/metric/prometheus` that could cause a panic. (#3899)
+- Fix sending nil `scopeInfo` to metrics channel in `go.opentelemetry.io/otel/exporters/metric/prometheus` that could cause a panic in `github.com/prometheus/client_golang/prometheus` (#3917)
```

Comment:
Are you able to add a unit test for this scenario?

Reply:
@pellared Implemented test to reproduce this issue, in main branch it gives expected test failure (and prometheus panic):

```go
2023/04/28 14:04:16 label value "\xff\xfe\xfd" is not valid UTF-8
2023/04/28 14:04:16 "___" is not a valid label name for metric "__total"
--- FAIL: TestNotNilScopeinfoInCollect (0.00s)
    exporter_test.go:798: 
        	Error Trace:	/tmp/opentelemetry-go/exporters/prometheus/exporter_test.go:798
        	Error:      	Expected value not to be nil.
        	Test:       	TestNotNilScopeinfoInCollect
FAIL
FAIL	go.opentelemetry.io/otel/exporters/prometheus	0.010
```